src/client/components/pages/Courses/RoomSelectionTable.tsx

```diff
@@ -61,19 +73,56 @@ const displayAvailability = (
 const RoomSelectionTable = (
   { roomList, addButtonHandler, currentRoomId }: RoomSelectionTableProps
 ): ReactElement<RoomSelectionTableProps> => {
+  /**
+   * The current campus filter value
+   */
+  const [
+    campusFilter,
+    setCampusFilter,
+  ] = useState<CAMPUS>(CAMPUS.ALL);
+
+  /**
+   * The current room filter value
+   */
+  const [
+    roomFilter,
+    setRoomFilter,
+  ] = useState<string>('');
+
+  /**
+   * The current availability filter value
+   */
   const [
     availabilityFilter,
     setAvailabilityFilter,
   ] = useState<AVAILABILITY>(AVAILABILITY.ALL);
+
+  /**
+   * Return filtered rooms based on the campus, room, and availability filters
+   */
+  const filteredRooms = (): RoomResponse[] => {
```

Comment:
Rather than define this as a function and call it in the rendered body of the table, can we wrap this in `useMemo` and then pass `filteredRooms` down directly? e.g.

```ts

const filteredRooms = useMemo(() => {
 ...
}, [roomList, campusFilter, roomFilter]);
```

Reply:
Good idea! This was updated in [7587a79](https://github.com/seas-computing/course-planner/pull/514/commits/7587a797ed82e3a34c19487d416dbaecfe27f612).